File: Program/Product.java
/* 
    Saya Tattha Maharany Yasmin Akbar dengan NIM 2201805 mengerjakan soal Latihan 2 
    dalam Praktikum mata kuliah Desain dan Pemrograman Berbasis Objek, untuk keberkahan-Nya
    maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamin.
*/
// Class product merupakan child dari clothing, dan grandchild dari shirt
class Product extends Clothing
{
    // atribut private yang dibutuhkan
    private String idProduct;
    private String name;
    private String brand;
    private int price;

    // konstruktor
    public Product() {

    }
    public Product(String idProduct, String name, String brand, int price)
    {
        this.idProduct = idProduct;
        this.name = name;
        this.brand = brand;
        this.price = price;
    }

    // getter dan setter
    public String get_idProduct()
    {
        return this.idProduct;
    }
    public void set_idProduct(String idProduct)
    {
        this.idProduct = idProduct;
    }
    public String get_name()
    {
        return this.name;
    }
    public void set_name(String name)
    {
        this.name = name;
    }
    public String get_brand()
    {
        return this.brand;
    }
    public void set_brand(String brand)
    {
        this.brand = brand;
    }
    public int get_price()
    {
        return this.price;
    }
    public void set_price(int price)
    {
        this.price = price;
    }
}

File: CPP/Program/Product.cpp
/* 
    Saya Tattha Maharany Yasmin Akbar dengan NIM 2201805 mengerjakan soal Latihan 2 
    dalam Praktikum mata kuliah Desain dan Pemrograman Berbasis Objek, untuk keberkahan-Nya
    maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamin.
*/

// library
#include <bits/stdc++.h>
#include "Clothing.cpp"

using namespace std;

class Product : public Clothing {
    private:
        string id_product, name, brand;
        int price;
    public:
        Product(){

        }   

    Product(string id_product, string name, string brand, int price)
    {
        this->id_product = id_product;
        this->name = name;
        this->brand = brand;
        this->price = price;
    } 

    string get_id_product()
    {
        return this->id_product;
    }
    void set_id_product(string id_product)
    {
        this->id_product = id_product;
    }
    string get_name()
    {
        return this->name;
    }
    void set_name(string name)
    {
        this->name = name;
    }
    string get_brand()
    {
        return this->brand;
    }
    void set_brand(string brand)
    {
        this->brand = brand;
    }
    int get_price()
    {
        return this->price;
    }
    void set_price(int price)
    {
        this->price = price;
    }

    ~Product(){};
};